app/src/main/java/com/example/tasky/agenda_details/data/MimeUtils.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.tasky.agenda_details.data
+
+import android.content.ContentResolver
+import android.net.Uri
+import android.webkit.MimeTypeMap
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+
+object MimeUtils {
+    fun getMimeType(contentResolver: ContentResolver, uri: String): String? {
+        val type = contentResolver.getType(Uri.parse(uri))
+        return type
+    }
+
+    fun getExtensionFromMimeType(mimeType: String?): String? {
+        return MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType)
+    }
+
+    fun toRequestBody(byteArray: ByteArray, mimeType: String?): RequestBody {
+        return byteArray.toRequestBody(mimeType?.toMediaTypeOrNull())
+    }
```

Comment:
This function is not really the job of a MimeUtils class. MimeTypes themselves have nothing to do with request bodies, that's just how you use the mime type in this specific case. Instead, I'd recommend to create the request body in your repository while using the functions from this MimeUtils class there.